server/server.go

```diff
@@ -1993,10 +1993,16 @@ func (s *Server) checkPathScope(ctx context.Context, txn storage.Transaction, pa
 	}
 
 	spath := strings.Trim(path.String(), "/")
```

Comment:
💭 Do we need to worry about `foo/../bar` etc, or do these either not work at all, or are they cleaned before reaching this?

Reply:
I think that's handled by the router before this, but it's an excellent test case regardless - will add 👍

Reply:
Indeed, it's handled before this - so if we do PUT on `/data/a/../a/b` it will respond with a redirect to `/data/a/b` which would be blocked. Testing turned out to be a bit hairy here since we'd need to change the behavior of the test client "globally" to follow redirects so I'll drop that for now.